### ISSUE TITLE
LMDB: pdnsutil edit-zone fails

### ISSUE THREAD

**Issue by zeha** [PowerDNS/pdns#8134]
- Program: Authoritative
 - Issue type: Bug report

### Short description

edit-zone appears to fail with lmdbbackend.

### Environment

 - Operating system: Debian sid
 - Software version: 4.2.0-rc2
 - Software source: "compiled yourself"

### Steps to reproduce

```
% sudo pdnsutil create-zone exa.org
Creating empty zone 'exa.org'
% sudo pdnsutil create-zone exa.org
% sudo pdnsutil list-all-zones            
exa.org
% sudo pdnsutil edit-zone exa.org
Checked 3 records of 'exa.org', 0 errors, 0 warnings.
Detected the following changes:
+exa.org 300 IN NS ns1.exa.org
+ns1.exa.org 300 IN A 127.0.0.1

(a)pply these changes, (e)dit again, (r)etry with original zone, (q)uit: y

(a)pply these changes, (e)dit again, (r)etry with original zone, (q)uit: a
Error: Duplicate RO transaction
```

NB: pdns-server was stopped when this was attempted.

**Comment by zeha** [comment]
The same can be observed in `add-record`:

```
% sudo pdnsutil add-record exa.org exa.org NS 100 ns1.exa.org
Error: Duplicate RO transaction
```

**Comment by Habbie** [comment]
I've bisected the `add-record` failure to:
```
commit f43646f5156f93f710671fbaeadaee4102dfb02d
Author: Peter van Dijk <peter.van.dijk@powerdns.com>
Date:   Fri Jun 7 16:29:37 2019 +0200

    auth API, pdnsutil: improve backend transaction correctness
```

:cry: 

This was #7891. Verschlimmbesserung!

**Comment by zeha** [comment]
Looks like the passing -1 to startTransaction is not a good idea with lmdbbackend...

`LMDBBackend::replaceRRSet` checks the domain_id against the `d_transactiondomainid`, but I think it shouldn't do that. If I'm not mistaken the API might also submit changes to multiple domains in a single TX.

**Comment by Habbie** [comment]
Unsurprisingly, it's also broken on `rel/auth-4.2.x`.

**Comment by Habbie** [comment]
> Looks like the passing -1 to startTransaction is not a good idea with lmdbbackend...

For `add-record`, that is not the problem. The 'problem' is that pdnsutil wants to both read and write in the same transaction, and the lmdbbackend does not support that (this is not by any means an LMDB limitation).

**Comment by Habbie** [comment]
This appears to be the main thing holding up 4.2.0(-rc3) now, so I'm postponing this issue, as sad as that makes me, so that the rest of 4.2.0 can go out the door.

**Comment by Habbie** [comment]
fixed in #8536 (master) and #8568 (4.2.x)

**Comment by zeha** [comment]
A doc update for https://doc.powerdns.com/authoritative/backends/lmdb.html (which has a big warning linking to this bug) would be good, I guess?
